src/Microsoft.NET.Sdk.Functions.Generator/MethodInfoExtensions.cs

```diff
@@ -59,15 +60,20 @@ public static JObject ManualTriggerBinding(this MethodDefinition method)
         /// <returns><see cref="FunctionJsonSchema"/> object that represents the passed in <paramref name="method"/>.</returns>
         public static FunctionJsonSchema ToFunctionJson(this MethodDefinition method, string assemblyPath)
         {
+            // For every SDK parameter, convert it to a FunctionJson bindings.
+            // Every parameter can potentially contain more than 1 attribute that will be converted into a binding object.
+            var bindingsFromParameters = method.HasNoAutomaticTriggerAttribute() ? new[] { method.ManualTriggerBinding() } : method.Parameters
+                                            .Select(p => p.ToFunctionJsonBindings())
+                                            .SelectMany(i => i);
+
+            // Get binding if a return attribute is used.
+            // Ex:  [return: Queue("myqueue-items-a", Connection = "MyStorageConnStr")]
+            var returnBindings = GetOutputBindingsFromReturnAttribute(method);
+            var allBindings = bindingsFromParameters.Concat(returnBindings);
```

Comment:
Let's retain the ToArray so the result is materialized. I don't know how downstream callers are using this.

Reply:
Ok. Kept the `ToArray` call as suggested. But as you already have noticed, the [Binding property ](https://github.com/Azure/azure-functions-vs-build-sdk/blob/main/src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonSchema.cs#L11)(to which this collection was assigned) is IEnumerable.

Reply:
That's fine - we just don't want a deferred enumerable that gets reevaluated each time